cpp/src/arrow/adapters/orc/adapter.h

```diff
@@ -47,6 +48,16 @@ struct StripeInformation {
   int64_t first_row_id;
 };
 
+/// \brief Version provided by the official ORC C++ library.
+struct ARROW_EXPORT OrcVersion {
+  int major;
+  int minor;
+  int patch;
+
+  /// \brief Return the current version of ORC C++ library.
+  static std::optional<OrcVersion> Get();
```

Comment:
Assuming the ORC_VERSION macro always exists, shouldn't we ensure that this always returns successfully?